pkg/util/inplaceupdate/inplace_utils.go

```diff
@@ -36,7 +39,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var inPlaceUpdatePatchRexp = regexp.MustCompile("^/spec/containers/([0-9]+)/image$")
+var (
+	inPlaceUpdatePatchRexp = regexp.MustCompile("^/spec/containers/([0-9]+)/image$")
+	UpdateExpectations     = expectations.NewUpdateExpectations(&revisionAdapterImpl{})
```

Comment:
`inplaceupdate` is a public package, don't put `UpdateExpectations`/`revisionAdapterImpl` of daemonset in here.